habitTrackerApp.py

```diff
@@ -155,26 +154,31 @@ def set_date(self, widget, *args, **kwargs):
         
         self.populate_header()
 
-        print('we are in set_date method')
         print(widget.background_color)
         print(MyBoxLayout.chosenColor)
+
         if widget.background_color == MyBoxLayout.chosenColor:
             print('we are in white color')
             widget.background_color = [1,1,1,1]
         else:
             print('color is: ' + str(MyBoxLayout.chosenColor))
             widget.background_color = MyBoxLayout.chosenColor
 
-        #Bookmark 25.01.2022: find out why the label text isn't updating automatically.
         counter = 0
         for dates in reversed(MyBoxLayout.buttonList[:MyBoxLayout.buttonList.index(MyBoxLayout.TheTodayButton)+1]):
             if dates.otherButtonID == date.today() and dates.background_color != MyBoxLayout.chosenColor:
                 continue
             elif dates.background_color == MyBoxLayout.chosenColor:
```

Comment:
it seems like you are using the date background color to identify some kind of state. 
It's best to use a proper identifier as things like color can change in the future.